File: src/main/java/lk/ijse/vehicleservice/entity/Vehicle.java
package lk.ijse.vehicleservice.entity;

import lombok.*;
import javax.persistence.*;


@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
@Entity
public class Vehicle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String vehicle_brand;
    @Column(nullable = false)
    private String vehicle_category;
    @Column(nullable = false)
    private String fuel_type;
    @Column(nullable = false)
    private String hybrid_OR_nonHybrid;
    @Column(nullable = false)
    private String fuel_usage;
    @Column(nullable = false)
    @Lob
    private byte[]vehicle_img;
    @Column(nullable = false)
    private int seatCapacity;
    @Column(nullable = false)
    private String  vehicle_type;
    @Column(nullable = false)
    private String driver_name;
    @Column(nullable = false)
    private String driver_contact;
    @Column(nullable = false)
    @Lob
    private byte[] driver_license_image;

}

File: Vehicle-FrontEnd/src/js/vehicle.js
loadAllVehicles();

function loadAllVehicles() {

    function loadHotelData() {
        $.ajax({
            url: 'http://localhost:8080/Mapping/api/vehicle', // Replace with your actual backend URL
            method: 'GET',
            dataType: 'json',
            success: function (data) {
                if (data.code === '200') {
                    var vehicles = data.data;
                    var vehiclelList = $('#card-container');

                    // Iterate through the hotel data and create card elements
                    for (var i = 0; i < vehicles.length; i++) {
                        var vehicle = vehicles[i];
                        var card = `
                                      <div class="col-12 col-md-6 col-lg-3 card-wrapper mt-4">
                                        <div class="card">
                                          <img class="card-img-top" src="images/1.jpg" alt="...">
                                          <div class="card-body">
                                            <!--  Customize this section  -->
                                             <div class="card-title" title=""><u><h5>`+vehicle.vehicle_brand+`</h5></u></div>
                                              <div class="card-title" title="">`+vehicle.vehicle_category+`</div>
                                              <div class="card-title" title="">`+vehicle.fuel_type+`</div>
                                              <div class="card-title" title="">`+vehicle.hybrid_OR_nonHybrid+`</div>
                                              <div class="card-title" title="">`+vehicle.fuel_usage+`</div>
                                              <div class="card-title" title="">`+vehicle.seatCapacity+`</div>
                                              <div class="card-title" title="">`+vehicle.vehicle_type+`</div>
                                              <div class="card-title" title="">`+vehicle.driver_name+`</div>
                                              <div class="card-title" title="">`+vehicle.driver_contact+`</div>

                                            <div style="float: right"><small>Free guid</small></div>
                                            <!--  End  -->
                                          </div>
                                        </div>
                                      </div>
                        `;



                        vehiclelList.append(card);

                        $("#card-container .col-12:last-child div img").attr('src', `data:image/png;base64,`+vehicle.vehicle_img+``);
                        $("#card-container .col-12:last-child div img").attr('src', `data:image/png;base64,`+vehicle.driver_license_image+``);
                    }
                } else {
                    // Handle the error case here
                    console.log('Error: ' + data.message);
                }
            },
            error: function (error) {
                // Handle the error case here
                console.log('Error: ' + error);
            }
        });
    }

    // Call the function to load hotel data
    loadHotelData();



}

